### PR TITLE
Issue12 product page listing

### PR DESCRIPTION
**Pull request by ninetteadhikari**
- GET request from /api/products and lists the products
- Search filter for name and tags
- Category filter
- Basic Material UI design

### CODE REVIEW

File: client/src/components/ProductsList.js [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+  render() {
+    const useStyles = makeStyles(theme => ({
+      card: {
+        maxWidth: 100
+      }
+    }));
+
+    const classes = useStyles;
+
+    const distinctCategory = [
```
Comment (1000miles) [inline client/src/components/ProductsList.js:61]: Can we comment it as a `dropdown` here?

File: client/src/components/ProductsList.js [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+                        color="textSecondary"
+                        component="p"
+                      >
+                        {data.description}
+                      </Typography>
+                    </CardContent>
+                  </CardActionArea>
+                  <CardActions>
+                    <Button size="small" color="primary">
+                      Share
```
Comment (1000miles) [inline client/src/components/ProductsList.js:129]: We discusseed this, it's just a placeholder for now.

File: client/src/components/ProductsList.js [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+                    </Button>
+                    <Button size="small" color="primary">
+                      Learn More
```
Comment (1000miles) [inline client/src/components/ProductsList.js:132]: We discusseed this, it's just a placeholder for now.

File: client/src/components/ProductsList.js [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+    });
+
+    return (
+      <>
+        <h1>Product List</h1>
+        <SearchFilter
+          searchText={this.state.searchText}
+          searchCategory={this.state.searchCategory}
+          handleChange={this.handleChange}
+          filteredProduct={filteredProduct}
```
Comment (1000miles) [inline client/src/components/ProductsList.js]: We discussed this, will be deleted because not used.

**Review by 1000miles — CHANGES_REQUESTED** [review]
Good job, @ninetteadhikari and @noliko-zwa ! 🍬Merci beaucoup. At the moment I can't test your branch without testing data. We probably would need to fix and merge PR #19 first. Except from minor change requests that we discussed yesterday I think we can merge soon.

File: client/src/components/ProductsList.js (hunk 1 above)
Comment (ninetteadhikari) [reply client/src/components/ProductsList.js:61]: added comment

File: client/src/components/ProductsList.js (hunk 2 above)
Comment (ninetteadhikari) [reply client/src/components/ProductsList.js:129]: added comment

File: client/src/components/ProductsList.js (hunk 3 above)
Comment (ninetteadhikari) [reply client/src/components/ProductsList.js:132]: added comment

File: client/src/components/ProductsList.js (hunk 4 above)
Comment (ninetteadhikari) [reply client/src/components/ProductsList.js]: deleted the extra prop

File: client/src/components/ProductsList.js (hunk 1 above)
Comment (1000miles) [reply client/src/components/ProductsList.js:61]: Thanks, looks good. 🍦

File: client/src/components/ProductsList.js (hunk 3 above)
Comment (1000miles) [reply client/src/components/ProductsList.js:132]: Thanks. :)

**Review by 1000miles — APPROVED** [review]
Merci beaucoup for your great work! 🍡

**Pull request merged**